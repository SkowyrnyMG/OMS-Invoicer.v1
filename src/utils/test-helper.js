import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
