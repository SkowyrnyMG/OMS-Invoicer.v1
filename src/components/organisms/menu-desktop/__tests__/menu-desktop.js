import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import MenuDesktop from '../menu-desktop';

describe('MenuDesktop', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<MenuDesktop />);
  });
});
