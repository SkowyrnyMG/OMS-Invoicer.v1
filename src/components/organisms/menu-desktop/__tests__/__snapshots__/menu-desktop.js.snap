// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDesktop should display in the document with correct styles 1`] = `
.c2 {
  display: inline-block;
  padding: 1rem 2rem;
  margin-top: 0;
  font-weight: 700;
  color: #000 !important;
  border-left: 1px solid #F0F0FD;
  background: none;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  border: none;
}

.c2:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 20px;
  font-weight: 700;
}

.c4 {
  width: 6rem;
}

.c1 {
  width: 100%;
  min-height: 100vh;
  grid-row: 1 / -1;
  display: grid;
  grid-template-rows: 6rem 1fr;
  background: #FFFFFF;
  z-index: 1000;
  overflow-y: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 5rem;
}

.c6 .active-app-link {
  color: #446DF6 !important;
  fill: currentColor !important;
  font-weight: bold;
}

.c7 {
  width: 85%;
  list-style: none;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 svg {
  margin-right: 1.3rem;
  width: 2.4rem !important;
}

.c9 {
  display: inline-block;
  width: 100%;
}

.c10 {
  display: inline-block;
  width: calc(100% - 5rem);
  height: 6rem;
  border-bottom: 1px solid #F0F0FD;
  line-height: 6rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: inherit;
  margin: 2rem 0;
  font-weight: 700;
  color: #F25C54;
  background: none;
  border: none;
  border-left: 1px solid #F0F0FD;
  cursor: pointer;
  -webkit-transition: 0.25s -webkit-transform;
  -webkit-transition: 0.25s transform;
  transition: 0.25s transform;
}

.c11 *:not(:last-child) {
  margin-right: 1rem;
}

.c11:hover {
  -webkit-transform: translateX(5px);
  -ms-transform: translateX(5px);
  transform: translateX(5px);
}

.c12 {
  fill: #F25C54;
}

.c0 {
  position: fixed;
  width: 15%;
  height: 100%;
  box-shadow: 4px 0 5px rgba(0,0,0,0.1);
  z-index: 1000;
}

@media (max-width:1300px) {
  .c5 {
    display: none;
  }
}

@media (max-width:1168px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/"
      linktype=""
    >
      <span
        class="c3"
      >
        <svg
          class="c4"
        >
          logo.svg
        </svg>
        <span
          class="c5"
        >
          Invoicer.v1
        </span>
      </span>
    </a>
    <nav
      class="c6"
    >
      <ul
        class="c7"
      >
        <li
          class="c8"
        >
          <a
            class="c9"
            data-testid="maindesk-link"
            href="/app"
          >
            <svg>
              home-icon.svg
            </svg>
            <span
              class="c10"
            >
              Main desk
            </span>
          </a>
        </li>
        <li
          class="c8"
        >
          <a
            class="c9"
            data-testid="invoices-link"
            href="/app/invoices"
          >
            <svg>
              invoice-icon.svg
            </svg>
            <span
              class="c10"
            >
              Invoices
            </span>
          </a>
        </li>
        <li
          class="c8"
        >
          <a
            class="c9"
            data-testid="orders-link"
            href="/app/orders"
          >
            <svg>
              orders-icon.svg
            </svg>
            <span
              class="c10"
            >
              Orders
            </span>
          </a>
        </li>
        <li
          class="c8"
        >
          <a
            class="c9"
            data-testid="customers-link"
            href="/app/customers"
          >
            <svg>
              clients-icon.svg
            </svg>
            <span
              class="c10"
            >
              Customers
            </span>
          </a>
        </li>
      </ul>
      <ul
        class="c7"
      >
        <li
          class="c8"
        >
          <a
            class="c9"
            data-testid="settings-link"
            href="/app/settings"
          >
            <svg>
              settings-icon.svg
            </svg>
            <span
              class="c10"
            >
              Settings
            </span>
          </a>
        </li>
        <li
          class="c8"
        >
          <a
            class="c9"
            data-testid="help-link"
            href="/app/help"
          >
            <svg>
              help-icon.svg
            </svg>
            <span
              class="c10"
            >
              Help
            </span>
          </a>
        </li>
        <button
          class="c11"
        >
          <svg
            class="c12"
          >
            logout-icon.svg
          </svg>
          <span>
            Logout
          </span>
        </button>
      </ul>
    </nav>
  </div>
</div>
`;
